clang/lib/CodeGen/Targets/NVPTX.cpp

```diff
@@ -259,6 +259,22 @@ void NVPTXTargetCodeGenInfo::setTargetAttributes(
         addNVVMMetadata(F, "maxntidx", MaxThreads);
         HasMaxWorkGroupSize = true;
       }
+    } else if (auto Attr = F->getFnAttribute("sycl-max-work-group-size");
+               Attr.isValid()) {
+      // Split values in the comma-separated list integers.
+      SmallVector<StringRef, 3> ValStrs;
```

Comment:
There's a lot of code overlap between NVPTX lowering of kernel properties and `CompileTimePropertiesPass` lowering of kernel properties. We might want to think of a shared utility to work with kernel properties.

Reply:
Hiding the fact that they're all represented as strings in LLVM IR as an "implementation detail" and providing a `std::optional<T> KernelPropertyAttr::getAsInteger` or `getAsIntegerList` might be a good step.